frontends/ios/BitBoxApp/BitBoxApp/Bluetooth.swift

```diff
@@ -151,21 +152,21 @@ class BluetoothManager: NSObject, ObservableObject, CBCentralManagerDelegate, CB
             return nil
         }
         print("Bluetooth wants to read \(length)")
-        readBufferLock.lock()
-        readBuffer.removeAll() // Clear buffer before starting
-        readBufferLock.unlock()
+
+        let count = length/64
+        
+        var data = Data()
 
         // Loop until we've read the required amount of data
-        while readBuffer.count < length {
-            // Trigger a read request
-            peripheral.readValue(for: pReader)
-            // Block until the delegate signals
+        for _ in 0..<count {
+            // Block until we have received `count` packets
             semaphore.wait()
+            readBufferLock.lock()
+            data.append(readBuffer.prefix(64))
+            readBuffer = readBuffer.advanced(by:64) // Clear buffer
```

Comment:
Hmm no auto-formating? space after colon etc.

Reply:
Enabled the LSP